docker/docker_image_src.go

```diff
@@ -129,9 +129,42 @@ func (s *dockerImageSource) ensureManifestIsLoaded() error {
 	return nil
 }
 
+func (s *dockerImageSource) getExternalBlob(urls []string) (io.ReadCloser, int64, error) {
+	var (
+		resp *http.Response
+		err  error
+	)
+	for _, url := range urls {
+		resp, err = s.c.makeRequestToResolvedURL("GET", url, nil, nil, -1)
```

Comment:
Yeah, using the certs but not sending any tokens/passwords is AFAICT safe.

Ultimately, we would need to decide where the tokens/passwords are configured, and whether that is a per-transport matter or something general, and so on, but if we don’t have any server needing this right now, let's get this merged first and worry about that later.